File: android/src/main/java/top/huic/tencent_im_plugin/message/LocationMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.v2.V2TIMLocationElem;
import com.tencent.imsdk.v2.V2TIMManager;
import com.tencent.imsdk.v2.V2TIMMessage;

import top.huic.tencent_im_plugin.message.entity.LocationMessageEntity;

/**
 * 位置消息节点
 */
public class LocationMessageNode extends AbstractMessageNode<V2TIMLocationElem, LocationMessageEntity> {
    @Override
    public V2TIMMessage getV2TIMMessage(LocationMessageEntity entity) {
        return V2TIMManager.getMessageManager().createLocationMessage(entity.getDesc(), entity.getLongitude(), entity.getLatitude());
    }

    @Override
    public String getNote(V2TIMLocationElem elem) {
        return "[位置消息]";
    }

    @Override
    public LocationMessageEntity analysis(V2TIMLocationElem elem) {
        return new LocationMessageEntity(elem);
    }

    @Override
    public Class<LocationMessageEntity> getEntityClass() {
        return LocationMessageEntity.class;
    }
}

File: ios/Classes/message/LocationMessageNode.swift
import ImSDK

//
//  TextMessageNode.swift
//  tencent_im_plugin
//
//  Created by 蒋具宏 on 2020/3/13.
//  位置消息节点
public class LocationMessageNode: AbstractMessageNode {

    override func getV2TIMMessage(params: [String: Any]) -> V2TIMMessage {
        let desc: String = getParam(params: params, paramKey: "desc")!;
        let longitude: Double = getParam(params: params, paramKey: "longitude")!;
        let latitude: Double = getParam(params: params, paramKey: "latitude")!;
        return V2TIMManager.sharedInstance().createLocationMessage(desc, longitude: longitude, latitude: latitude)
    }

    override func getNote(elem: V2TIMElem) -> String {
        "[位置消息]";
    }

    override func analysis(elem: V2TIMElem) -> AbstractMessageEntity {
        LocationMessageEntity(elem: elem as! V2TIMLocationElem);
    }
}
